### ISSUE TITLE
unable to establish connection with r session

### ISSUE THREAD

**Issue by galafab** [rstudio/rstudio#9105]
I can't use rstudio anymore because of this and other errors. 
1) Many times it doesn't even open it, it says "R is taking longer than usual to connect"
2)Most of the times I try to do something, even some basic thing like 1+1, or a<-1, the message appears "unable to establish connession with R session"
3) Sometimes another message appear "RStudio was unable to autosave this file. You may need to restart RStudio".
I don't know what to do, I tried reinstalling it, I checked if I had the antivirus software(Avast) but i don't have it, I used ctrl+open file to select the R version to start RStudio, but nothing worked.

**Comment by mikebessuille** [comment]
We need more details - which version of RStudio?   Please fill out the standard bug template when you file an issue.  Have you tried running R from the console rather than within RStudio - does that work?

**Comment by galafab** [comment]
I tried running R from the console and it works.
Rstudio version is 1.4.1106
I tried with the previous version too but it still doesn't work(version 1.3.1093)

**Comment by augustohassel** [comment]
same here.. from one day to another it stoped working. 

`R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.1.0 tools_4.1.0 `

RStudio version: 1.4.1717

I've tried downgrading, reinstalling, etc..

**Comment by colearendt** [comment]
This seems problematic if still at large

**Comment by mikebessuille** [comment]
I'm looking for a support article on the topic.  This is an uncommon problem but it does come up, and there are several key things to try to resolve it:
1) Reset state per https://support.rstudio.com/hc/en-us/articles/200534577-Resetting-RStudio-Desktop-s-State
2) Disable antivirus
3) change to software rendering
4) ensure your working directory is local, not in a cloud folder (e.g., Onedrive, etc)

**Comment by mikebessuille** [comment]
Internal:  See https://docs.google.com/document/d/1Usih1DbYvE7qtRbTqeqtcisgSut5UZfNtL8pyrIif6o/edit#.  We are going to write a definitive support guide on this problem, detailing the many possible causes and potential solutions.  This is often caused by user code which deadlocks or runs a long loop that doesn't let R communicate back to the IDE for a long time.

**Comment by colearendt** [comment]
Not yet, stalebot!

**Comment by mikebessuille** [comment]
I don't see enough detail in the current issue for us to be able to take action on this.  I suggest updating to the latest, and reviewing https://support.rstudio.com/hc/en-us/articles/4423821070743-Troubleshooting-RStudio-IDE-Connection-Errors for potential resolutions.  If you are still having trouble, please open a new issue with enough information following the bug template; e.g, whether desktop or server version, which version, running from the command line or not, how did you install it (installer, built it yourself, etc...)

**Comment by mjbroerman** [comment]
After reinstalling R and RStudio, I am still having this issue, if you want to make use of my state, it may not last for long (might nuke my OS if it comes down to it)!

Interestingly, I can still do a number of things: compile with knitr in .rmd and .qmd, load a few objects. But after a while loading and operating on objects, the console hangs, saving no longer works, and I got the autosaving error at the top, which is how I got here.
